### PR TITLE
Divided Update Changes into separate rename/delete changes

### PR DESCRIPTION
**Pull request by oberlehner**
Delete Changes should not change anything, therefore they are disabled per default.
Rename Changes need to actually rename the regarding element as default option, so it is not possible to use the same changes for both refactorings.
Therefore, they are splitted up which also gives the benefits to provide more specific repair options.

Furthermore, if a change is enabled, there will be always a selection in the viewer, so it is not possible for the user to enable a change without a corresponding operation that needs to exectuted

### CODE REVIEW

File: plugins/org.eclipse.fordiac.ide.typemanagement/src/org/eclipse/fordiac/ide/typemanagement/Messages.java [hunk 1]
```diff
@@ -113,6 +113,8 @@ public final class Messages extends NLS {
 	public static String DeleteFBTypeParticipant_Change_UpdateManipulator;
 	public static String DeleteFBTypeParticipant_Change_UpdateSubappPins;
 	public static String FBTypeComposedAdapterFactory_FBTypecomposedAdapterFactoryShouldNotBeInsantiated;
+	public static String IFordiacPreviewChange_0;
```
Comment (mx990) [inline plugins/org.eclipse.fordiac.ide.typemanagement/src/org/eclipse/fordiac/ide/typemanagement/Messages.java]: Please change the name of the constant to something meaningful.

File: plugins/org.eclipse.fordiac.ide.typemanagement/src/org/eclipse/fordiac/ide/typemanagement/refactoring/rename/RenameUpdateFBTypeInterfaceChange.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+	public RenameUpdateFBTypeInterfaceChange(final FBType type, final StructuredType struct) {
+		super(MessageFormat.format(Messages.DeleteFBTypeParticipant_Change_DeleteFBTypeInterface, type.getName(),
+				struct.getName()), EcoreUtil.getURI(type), FBType.class);
+		this.struct = struct;
+	}
+
+	@Override
+	public EnumSet<ChangeState> getAllowedChoices() {
+		return EnumSet.of(ChangeState.RENAME, ChangeState.CHANGE_TO_ANY, ChangeState.DELETE);
```
Comment (mx990) [inline plugins/org.eclipse.fordiac.ide.typemanagement/src/org/eclipse/fordiac/ide/typemanagement/refactoring/rename/RenameUpdateFBTypeInterfaceChange.java]: Should we even offer the choices `CHANGE_TO_ANY` and `DELETE` when renaming?

File: plugins/org.eclipse.fordiac.ide.typemanagement/src/org/eclipse/fordiac/ide/typemanagement/refactoring/rename/RenameUpdateStructDataTypeMemberVariableChange.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+	public RenameUpdateStructDataTypeMemberVariableChange(final VarDeclaration varDeclaration) {
+		super(MessageFormat.format(Messages.DeleteFBTypeParticipant_Change_UpdateMemberVariable,
+				varDeclaration.getName(), varDeclaration.getTypeName(),
+				((INamedElement) varDeclaration.eContainer()).getName()), EcoreUtil.getURI(varDeclaration),
+				VarDeclaration.class);
+	}
+
+	@Override
+	public EnumSet<ChangeState> getAllowedChoices() {
+		return EnumSet.of(ChangeState.RENAME, ChangeState.DELETE, ChangeState.CHANGE_TO_ANY, ChangeState.NO_CHANGE);
```
Comment (mx990) [inline plugins/org.eclipse.fordiac.ide.typemanagement/src/org/eclipse/fordiac/ide/typemanagement/refactoring/rename/RenameUpdateStructDataTypeMemberVariableChange.java]: same here

**Review by mx990 — CHANGES_REQUESTED** [review]
I would also suggest adding a fatal error in `isValid` for each of these changes when `NO_CHANGE` is selected, so the refactoring cannot proceed when any such changes are enabled.

File: plugins/org.eclipse.fordiac.ide.typemanagement/src/org/eclipse/fordiac/ide/typemanagement/refactoring/rename/RenameUpdateFBTypeInterfaceChange.java (hunk 2 above)
Comment (oberlehner) [reply plugins/org.eclipse.fordiac.ide.typemanagement/src/org/eclipse/fordiac/ide/typemanagement/refactoring/rename/RenameUpdateFBTypeInterfaceChange.java]: you are right, i think i will remove it unless someone explicitly wants it

**Review by mx990 — APPROVED** [review]

**Review by azoitl — APPROVED** [review]

**Pull request merged**